spyder/plugins/ipythonconsole.py

```diff
@@ -1079,11 +1099,38 @@ def create_new_client(self, give_focus=True, filename='', is_cython=False):
                                      "<tt>conda install ipykernel cloudpickle</tt>"))
                 return
 
-        self.connect_client_to_kernel(client, is_cython=is_cython)
+        self.connect_client_to_kernel(client, is_cython=is_cython,
+                                      is_pylab=is_pylab, is_sympy=is_sympy)
         if client.shellwidget.kernel_manager is None:
             return
         self.register_client(client)
 
+    def create_pylab_client(self, give_focus=True, filename='',
+                            is_cython=False, is_pylab=True, is_sympy=False):
+        """Force creation of pylab client"""
+        self.create_new_client(give_focus=give_focus,
+                               filename=filename,
+                               is_cython=is_cython,
+                               is_pylab=is_pylab,
+                               is_sympy=is_sympy)
+        # Rename client tab with Pylab
+        client = self.get_current_client()
+        tab_text = _("Pylab") + u' ' + client.id_['int_id']

```

Comment:
This is not the right way to do this. Please add a new kwarg to `create_new_client` called `given_name=None` and pass its value to `ClientWidget`. Then here you simply need to do

```python
console_name = _('Pylab')
self.create_new_client(is_pylab=True,
                       given_name=console_name)
```

and `ClientWidget` methods will take care of correctly naming the new console.

Reply:
Another correction, `console_name` needs to be `console_name = 'Pylab'` because `Pylab` can't be translated.

Reply:
I forgot to mention that `ClientWidget` has an arg called `given_name`.

Reply:
Using `given_name` seems to give me `SymPy/A` every time I open a new console instead of `SymPy 2/A` and 3 and 4 as I create more.  Should I change the behavior of `ClientWidget.get_name` to always include the console id so I can have the original behavior? 

The only other place I can see that uses `get_name` is for remote connections. I have never used remote sessions but if I opened 2 of the same connections I would still want them to automatically get unique name.

Also what's the `/A` for? It seem to be static so all local console get it.